pkg/app/bp.go

```diff
@@ -59,6 +60,12 @@ func (b AppBlueprint) Blueprint() *crv1alpha1.Blueprint {
 	if err != nil {
 		log.Error().WithError(err).Print("Failed to read Blueprint", field.M{"app": b.App})
 	}
+
+	// set the name to a dynamically generated value
+	// so that the name wont conflict with the same application
+	// installed using other ways
+	bpr.ObjectMeta.Name = fmt.Sprintf("%s-%s", bpr.ObjectMeta.Name, rand.String(5))
```

Comment:
were we not already doing this?

Reply:
we were.

Reply:
https://github.com/kanisterio/kanister/pull/1187/files#diff-3fd34ab36ccd8befaf7cfcd5a617f87923387b85543338ab8f18dee033be9d64L50-L54